### ISSUE TITLE
Automatically close/shutdown when torrents are finished/reaches a certain ratio

### ISSUE THREAD

**Issue by vktr** [picotorrent/picotorrent#227]
We should add an option to automatically close PicoTorrent or shutdown the computer when all torrents are finished or has reached a certain seeding ratio.

**Comment by LeBaux** [comment]
What is your stance on leechers and minimal seed ratio? If you make it possible/easy for users to seed nothing, most of them do just that.

**Comment by vktr** [comment]
Sorry for the late answer. My stance is that this should only work when a seed ratio has been achieved, for example only allow shutdown when reaching a ratio of at least 1 or 2.

**Comment by cubikca** [comment]
I think we need this feature as well. I'd like to download overnight occasionally without chewing up tons of bandwidth. I'd add, however, that it makes no sense to add this feature with such restrictions. Either you allow a target share ration to be defined, or you don't. It's not a developer's place to police or judge the behaviors of their users. In any case, it's open source software, so such restrictions are trivial to remove.

**Comment by vktr** [comment]
I agree that we shouldn't police the behavior of our users. I just think there's a collective responsibility to keep swarms healthy. That doesn't mean we shouldn't implement this feature, it's just food for thought 😄 

As for your specific issue, you could lower your upload speed, and perhaps a bandwidth scheduler would be more of use than removing torrents?

**Comment by eurobank** [comment]
@vktr Please let the user decide what he wants to do.
